scvi/dataset/dataset.py

```diff
@@ -241,28 +241,29 @@ def populate_from_datasets(
         For cell-wise attributes, either we "concatenate" or add an "offset" corresponding
         to the number of already existing categories.
 
-        :param gene_datasets: a sequence of ``GeneExpressionDataset`` objects.
-        :param on: attribute to select gene intersection
-        :param sharing_instructions_dict: Instructions on how to share cell-wise attributes between datasets.
-            Keys are the attribute name and values are either:
-                * "offset": to add an offset corresponding to the number of categories already existing.
-                e.g for batch_indices, if the first dataset has batches 0 and 1,
-                in the merged dataset, the second dataset's batch indices will start at 2.
-                * "concatenate": concatenate the attribute, no changes applied.
+        :param gene_datasets_list: ``GeneExpressionDataset`` objects to be merged.
+        :param shared_labels: whether to share labels through ``cell_types`` mapping or not.
+        :param mapping_reference_for_sharing: Instructions on how to share cell-wise attributes between datasets.
+            Keys are the attribute name and values are registered mapped attribute.
+            If provided the mapping is merged across all datasets and then the attribute is
+            remapped using index backtracking between the old and merged mapping.
         """
         # sanity check
-        if not all([hasattr(gene_dataset, on) for gene_dataset in gene_datasets_list]):
+        if not all([hasattr(gene_dataset, "gene_names") for gene_dataset in gene_datasets_list]):
             raise ValueError(
-                "All datasets should have the merge key 'on' as an attribute"
+                "All datasets should have a gene_names attribute."
             )
 
-        # set default sharing behaviour
-        if sharing_intstructions_dict is None:
-            sharing_intstructions_dict = {"batch_indices": "offset"}
+        # set default sharing behaviour for batch_indices and labels
+        if mapping_reference_for_sharing is None:
+            mapping_reference_for_sharing = {}
+        if shared_labels:
+            mapping_reference_for_sharing = {"labels": "cell_types"}
+        mapping_reference_for_sharing.update({"batch_indices": None})
```

Comment:
You typed the dictionary as `Dict[str, str]`. Here why do you have None?

Reply:
typo, we don't need this line as the call is ``mapping_reference_for_sharing.get(.., default=None)``

Reply:
Good